Packages/com.unity.inputsystem/InputSystem/State/InputStateBlock.cs

```diff
@@ -223,6 +223,67 @@ public int ReadInt(void* statePtr)
             return value;
         }
 
+        public void WriteInt(void* statePtr, int value)
+        {
+            Debug.Assert(sizeInBits != 0);
+
+            var valuePtr = (byte*)statePtr + (int)byteOffset;
+
+            if (format == FormatInt || format == FormatUInt)
```

Comment:
use a switch statement?

Reply:
`case` clauses have to be compile-time constant so can't use `FormatInt` etc. We could go and use plain ints instead but think that would make the code here quite a lot less readable.